presto-record-decoder/src/main/java/com/facebook/presto/decoder/avro/AvroColumnDecoder.java

```diff
@@ -189,13 +203,16 @@ private static Slice getSlice(Object value, Type type, String columnName)
     {
         switch (type.getTypeSignature().getBase()) {
             case VARCHAR:
-                if (value instanceof Utf8) {
+                if (type instanceof VarcharType && (value instanceof CharSequence || value instanceof GenericEnumSymbol)) {
```

Comment:
do you have any context about why we get rid of Utf8 here? or is there any test covered the VARCHAR type to make sure we're not breaking anything?

Reply:
`Utf8` implements `CharSequence` so it should be more compatible in situations where the Avro schema used a different class for `string` types.

Reply:
I see, thank you for the explanation!